pkg/client/client.go

```diff
@@ -73,9 +73,10 @@ type Client struct {
 	mclient               monitoring.Interface
 	eclient               apiextensionsclient.Interface
 	aggclient             aggregatorclient.Interface
+	ctx                   context.Context
 }
 
-func New(cfg *rest.Config, version string, namespace, userWorkloadNamespace string) (*Client, error) {
+func New(cfg *rest.Config, version string, namespace, userWorkloadNamespace string, ctx context.Context) (*Client, error) {
```

Comment:
by Go conventions `ctx context.Context` should be the first function parameter.

Reply:
Indeed, I wrongly recommended putting it after the client in the precedent PR, so [CreateSecret](https://github.com/openshift/cluster-monitoring-operator/blob/master/test/e2e/framework/secret.go#L29) and [NewPrometheusClientFromRoute](https://github.com/openshift/cluster-monitoring-operator/blob/master/test/e2e/framework/client.go#L46-L51) should be updated.